### PR TITLE
cosmic-design-demo: init at unstable-2024-01-08

### PR DESCRIPTION
**Pull request by nyabinary**
## Description of changes

Design Demo for COSMIC DE
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/by-name/co/cosmic-design-demo/package.nix [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+  postInstall = ''
+    wrapProgram "$out/bin/cosmic-design-demo" \
+      --suffix XDG_DATA_DIRS : "${cosmic-icons}/share"
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/pop-os/cosmic-design-demo";
+    description = "Design Demo for the COSMIC Desktop Environment";
+    license = licenses.gpl3Only;
```
Comment (nbdd0121) [inline pkgs/by-name/co/cosmic-design-demo/package.nix:68]: cosmic-design-demo is MPL 2.0

**Review by nbdd0121 — CHANGES_REQUESTED** [review]
I got this error when building:
```
error: The `!include` directive has been stabilized as `import`
```
The error is added in Just 1.18 (released 2023-12-24).

File: pkgs/by-name/co/cosmic-design-demo/package.nix (hunk 1 above)
Comment (nyabinary) [reply pkgs/by-name/co/cosmic-design-demo/package.nix]: fixed

**Comment by nyabinary** [comment]
> I got this error when building:
> 
> ```
> error: The `!include` directive has been stabilized as `import`
> ```
> 
> The error is added in Just 1.18 (released 2023-12-24).

I believe this is an error on COSMIC side?

**Comment by nbdd0121** [comment]
It was an unstable feature that cosmic-design-demo used, but the feature was stablised in just 1.17 and unstable version removed in just 1.18. cosmic-design-demo's justfile hasn't been updated in the meanwhile.

**Comment by nyabinary** [comment]
> It was an unstable feature that cosmic-design-demo used, but the feature was stablised in just 1.17 and unstable version removed in just 1.18. cosmic-design-demo's justfile hasn't been updated in the meanwhile.

i wonder if we can do a sed and replace  that line for now?

**Comment by nbdd0121** [comment]
I just tested it, but unfortunately that's not sufficient. `import` is different from `!include` in terms of working directory -- the former uses the directory that contains the included file, and the later uses that of the including file.

**Comment by nyabinary** [comment]
> I just tested it, but unfortunately that's not sufficient. `import` is different from `!include` in terms of working directory -- the former uses the directory that contains the included file, and the later uses that of the including file.

Maybe opening a PR for cosmic-design-demo and updating the justfile is the best option then.

**Comment by nyabinary** [comment]
> I just tested it, but unfortunately that's not sufficient. `import` is different from `!include` in terms of working directory -- the former uses the directory that contains the included file, and the later uses that of the including file.

Been fixed now :)

**Pull request merged**